eng/pipelines/templates/variables/vmr-stage.yml

```diff
@@ -0,0 +1,49 @@
+variables:
+- ${{ if ne(parameters.vmrBranch, '') }}:
```

Comment:
Take care, and please double check, we provide a space ` ` as a default value for some optional parameters and then replace it away. We do this to make the top-level pipeline parameters optional but empty by default. So just make sure where this template is embedded and whether it has been stripped away already

Reply:
You can also guard against this relatively cheaply with `${{ if notIn(parameters.vmrBranch, '', ' ') }}:`

Reply:
I checked and it is stripped away at this point